src/proxy/outbound.rs

```diff
@@ -78,11 +84,12 @@ impl Outbound {
         //
         // So use a drain to nuke tasks that might be stuck sending.
         let (sub_drain_signal, sub_drain) = drain::channel();
+        let pi = Arc::new(self.pi);
```

Comment:
See, this makes me wonder if ProxyInputs is just a heterogeneous bag of junk we should split up, or simply always wrap in an `Arc` (not just here).